twake/backend/node/src/services/workspaces/services/workspace/service.ts

```diff
@@ -514,4 +531,63 @@ export class WorkspaceService implements WorkspaceServiceAPI {
   getUsersCount(workspaceId: string): Promise<number> {
     return this.workspaceCounter.get(this.wsCountPk(workspaceId));
   }
+
+  private jwtInviteToken(companyId: string, workspaceId: string, token: string) {
+    const inviteTokenObject: InviteTokenObject = { c: companyId, w: workspaceId, t: token };
+    return this.auth.sign(inviteTokenObject);
+  }
+
+  async getInviteToken(
+    companyId: string,
+    workspaceId: string,
+  ): Promise<WorkspaceInviteTokenObject> {
+    const pk = { company_id: companyId, workspace_id: workspaceId };
+    const res = await this.workspaceInviteTokensRepository.findOne(pk);
+    if (!res) return null;
+
+    return {
+      token: this.jwtInviteToken(companyId, workspaceId, res.invite_token),
+    };
+  }
+
+  async createInviteToken(
+    companyId: string,
+    workspaceId: string,
+  ): Promise<WorkspaceInviteTokenObject> {
+    await this.deleteInviteToken(companyId, workspaceId);
+    const token = randomBytes(16).toString("hex");
```

Comment:
@romanesko you could improve security and/or size with a conversion to base64 instead of hex (for your next PRs)